server/index.js

```diff
@@ -1,4 +1,71 @@
 const express = require('express');
+const mysql = require('promise-mysql');
+const bodyParser = require('body-parser');
 
+let connection;
 const app = express();
 app.use(express.static('public'));
+app.use(bodyParser.json());
+
+app.get('/listings/:listingId/reviews/', (req, res) => {
+  mysql.createConnection({
+    host: '127.0.0.1',
+    user: 'root',
+    password: '',
+    database: 'etsycutioner',
+  }).then((conn) => {
+    // capture connection
+    connection = conn;
+    // retrieve all listings for shop
+    return connection.query('SELECT shop FROM listings WHERE listings.id = ?', req.params.listingId);
+  }).then((data) => {
+    // pick shop id out of data packet
+    const { shop } = data[0];
+    // return all reviews for shop
+    return connection.query('SELECT DISTINCT reviews.id, reviews.person, reviews.shop, reviews.listing, reviews.body, reviews.date, reviews.stars FROM reviews, listings WHERE listings.shop = ? AND listings.shop = reviews.shop', shop);
```

Comment:
1. The query definitely returns what I want it to: each review for the shop from the given listing, with all listed info for each review. I generated my data to make sure that reviews.shop and listings.shop would be consistent; the redundant shop datum in each review was Danny's idea.
I understood what I was doing to be an implicit inner join, without the explicit command; is that incorrect? Is there an advantage to doing it otherwise? 

2. I struggled a bit with building out a bigger query before breaking them up. Would you mind spelling out the logic for how to accomplish it in one query? I'm not sure how WHERE (SELECT…) will help if I need the shop id before the main query. Sorry for inviting a MySQL clinic here, haha.

Reply:
Ah subquery. Thanks for that